### DISCUSSION TITLE
Vaultwarden Minimum Hardware Requirements

### DISCUSSION THREAD

**Discussion by NoniDOTio** [dani-garcia/vaultwarden#2903] [General]
Hello friends, 
I've been running vaultwarden on my homeserver for quite a while and noticed that it consumers very little resources.
As I am currently looking into moving some of my stuff to Azure I was wondering what the smallest VM size would be to host a vaultwarden docker container on.

Or more general:
**What are the minimum hardware requirements for vaultwarden?**
Will it work on a 1 vCore 1 GB RAM debian machine or does it require multiple cores / more memory headroom?

**Comment by BlackDex** [comment]
It all depends a bit on how much traffic you have, how large your database is etc.. if all is little and small, then thats probably going to work. You might want to see if there are different results between the debian or alpine images. Or maybe even extract the binary from the alpine imagesz or build it your self.

That would save some docker resources. Podman could be an option also.

But in short, i think it will work.
Though, keep in mind that Azure uses CIFS/SMB storage (as far as i know), that could cause some issues with sqlite databases for example.

**Comment by bokkabonga** [comment]
One Core should probably be fine if you don´t have too much load on the server. 1GB of RAM could become a problem, especially if you should plan to build the container yourself.

I ran into some problems with this with 2GB of RAM, but only while building the container.

**Comment by jsievertde** [comment]
Hmm you could also just use the bare metal version.
That way you don't need the overhead of docker or podman.
https://github.com/JensTimmerman/ansible-role-vaultwarden
I'm running that on on core and 512 mb with no problems right now.
